src/StagedProposalProcessor.sol

```diff
@@ -204,13 +236,23 @@ contract StagedProposalProcessor is ProposalUpgradeable, PluginUUPSUpgradeable {
         });
     }
 
+    /// @inheritdoc IProposal
     function createProposal(
-        bytes calldata _metadata,
-        IDAO.Action[] calldata _actions,
+        bytes memory _metadata,
+        Action[] memory _actions,
         uint64 _startDate,
-        uint64 /** */
-    ) external override returns (uint256 proposalId) {
-        proposalId = uint256(createProposal(_metadata, _actions, 0, _startDate));
+        uint64 /** */,
+        bytes memory _data
+    ) public virtual override returns (uint256 proposalId) {
+        proposalId = uint256(
+            createProposal(_metadata, _actions, 0, _startDate, abi.decode(_data, (bytes[][])))
+        );
+    }
+
+    /// @inheritdoc IProposal
+    /// @dev Since SPP is also IProposal, it's required to override. Though, ABI can not be defined at compile time.
+    function createProposalParamsABI() external pure virtual override returns (string memory) {
+        return "";
```

Comment:
I think if we don't enforce this, devs might not even realize that this is important. I prefer to leave this required as this will make them think what it is and ask questions in case they have.

Reply:
https://github.com/aragon/staged-proposal-processor-plugin/pull/8/commits/4510a1a6f46fe4cb54483868695341adc74da253